dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/file/FileHandler.kt

```diff
@@ -57,10 +61,51 @@ internal class FileHandler {
             }
     }
 
+    fun writeData(
```

Comment:
Is the migration of this method from the `ImmediateFileWriter` to this class really necessary anymore? It would greatly simplify this PR to only change what's necessary.

Reply:
I would like to keep this refactor because I want the FileHandler to be responsible about all the file operations. That is the reason I mainly moved this method there (wanted to do it for a long time by now).